common/components/GasSlider/components/AdvancedGas.tsx

```diff
@@ -3,53 +3,78 @@ import classnames from 'classnames';
 import translate from 'translations';
 import { DataFieldFactory } from 'components/DataFieldFactory';
 import FeeSummary from './FeeSummary';
+import { Spinner } from 'components/ui';
+import { CSSTransition } from 'react-transition-group';
+import { State as NetworkState } from 'reducers/transaction/network';
 import './AdvancedGas.scss';
 
 interface Props {
+  gasEstimationStatus: NetworkState['gasEstimationStatus'];
+  setGasLimit: boolean;
   gasPrice: string;
   gasLimit: string;
   nonce: string;
+  toggleSetGasLimit(bool: boolean): void;
   changeGasPrice(gwei: string): void;
   changeGasLimit(wei: string): void;
   changeNonce(nonce: string): void;
 }
 
 export default class AdvancedGas extends React.Component<Props> {
   public render() {
-    // Can't shadow var names for data & fee summary
-    const vals = this.props;
-
+    // Shadow var names on gasLimit
+    const { gasEstimationStatus, setGasLimit, gasPrice, nonce } = this.props;
+    const estimatingGas = gasEstimationStatus === 'PENDING' ? true : false;
     return (
       <div className="AdvancedGas row form-group">
+        <div className="col-md-12">
+          <label className="checkbox">
+            <input
+              type="checkbox"
+              className="form-check-input"
+              defaultChecked={setGasLimit}
+              onChange={this.handleToggleSetGasLimit}
+            />
+            <span>Automatically Set Gas Limit</span>
```

Comment:
As with previous comments, I'd call this calculate or something. Also, now that we have an explicit checkbox, I think we should make the gasLimit input disabled while this is unchecked.